internal/controller/stages/stages.go

```diff
@@ -439,8 +441,13 @@ func newReconciler(
 	r := &reconciler{
 		kargoClient:  kargoClient,
 		argocdClient: argocdClient,
-		recorder:     recorder,
-		cfg:          cfg,
+		directivesEngine: directives.NewSimpleEngine(
+			nil, // TODO: is this required here?
```

Comment:
That nil is the creds db, right?

I don't want to rule out the idea that third-party directives might require access to credentials.

imho, the best bet is to instantiate just one engine in the controller startup logic and inject that into both of the reconcilers that use it.